src/canvas.ts

```diff
@@ -0,0 +1,177 @@
+import utils from './utils';
+import Polygon from './polygon';
+
+import ISetting from './interfaces/canvas-setting.interface';
+
+export default class Canvas {
+  id: string;
+  setting: ISetting;
+  objects: Polygon[];
+  selectedObject: Polygon;
+  element: any;
+  ctx: CanvasRenderingContext2D;
+  nextObjListPos: number[];
+
+  constructor(id: string, setting: ISetting, objects?: Polygon[]) {
+    this.id = id;
+    this.setting = setting;
+    this.objects = objects || [];
+    this.selectedObject = null;
+    this.element = document.getElementById(id);
+    this.ctx = this.element.getContext('2d');
+    this.nextObjListPos = [this.setting.padding, this.setting.padding];
+  }
+
+  public init(): any {
+    this.element.width = this.setting.width;
+    this.element.height = this.setting.height;
+  }
+
+  public add(object: Polygon): void {
+    this.objects.push(object);
+    object.setBoundingBox();
+
+    const offset = [this.nextObjListPos[0] - object.boundingBox[0][0],
+      this.nextObjListPos[1] - object.boundingBox[0][1]];
+
+    object.shift(offset);
+    object.setBoundingBox();
+    this.nextObjListPos[1] = object.boundingBox[object.boundingBox.length - 1][1] +
+      this.setting.polygonMargin;
+    this.update();
+  }
+
+  public addArr(objectArr: Polygon[]): void {
+    objectArr.forEach((object) => {
+      this.add(object);
+    });
+  }
+
+  public draw(object: Polygon, isFill?: boolean): void {
+    this.ctx.fillStyle = object.fillColor;
+    this.ctx.strokeStyle = object.strokeColor;
+
+    this.ctx.save();
```

Comment:
`save` should be above `fillStyle` / `strokeStyle` changes, otherwise they will remain changed